packages/angular/ssr/node/src/app-engine.ts

```diff
@@ -33,6 +33,38 @@ export interface AngularNodeServerAppManager {
    * (e.g., `./logo.png`) rather than an application route.
    */
   render(request: Request | IncomingMessage, requestContext?: unknown): Promise<Response | null>;
+
+  /**
+   * Retrieves the HTTP headers for a given request based on the URL pathname
+   * of statically generated (SSG) pages.
+   *
+   * @param request - The incoming request object. This can be either a `Request` object (web API)
+   *                  or an `IncomingMessage` object (Node.js).
+   * @returns A record of HTTP headers as a key-value pair object, or `undefined` if there are no
+   *          headers defined for the given pathname.
+   *
+   * @example
+   * ```typescript
+   * const angularAppEngine = getOrCreateAngularNodeAppEngine();
+   *
+   * app.use(express.static('dist/browser', {
+   *   setHeaders: (res, path) => {
+   *     // Retrieve headers for the current request
+   *     const headers = angularAppEngine.getHeaders(res.req);
+   *
+   *     if (!headers) {
+   *       return;
+   *     }
+   *
+   *     // Set the retrieved headers on the response
+   *     for (const [key, value] of Object.entries(headers)) {
+   *       res.setHeader(key, value);
+   *     }
+   *   }
+   * }));
+   * ```
+   */
+  getHeaders(request: Request | IncomingMessage): Readonly<Record<string, string>> | undefined;
```

Comment:
Consider: I assume this is `Readonly` because we're returning the existing instance from the manifest. Would it be worth cloning that object just to protect it? `readonly` in TS is not great at protecting data, it's trivially easy to cast out of, and I don't think a lot of developers respect it. Is this a strong enough guarantee that it won't be modified?